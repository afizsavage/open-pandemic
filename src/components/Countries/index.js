import React, { useEffect } from 'react';
import axios from 'axios';
import { useDispatch, useSelector } from 'react-redux';

import { ADD_COUNTRY, createAction } from '../../redux/countries/index';
import ListOfCountries from './countries-list';
import HeroSection from './hero-section';
import { baseAPI, getCurrentDate } from './api';

function importAllImages(r) {
  return r.keys().map(r);
}

const imageList = importAllImages(
  require.context('../../images', false, /\.(png|jpe?g|svg)$/),
);

const populateReduxStore = (data, dispatch) => {
  let img;
  const dataObj = Object.values(data)[0];
  const countries = Object.values(dataObj);

  countries.forEach((country) => {
    img = imageList.filter((image) => image.includes(country.id));
    dispatch(createAction(ADD_COUNTRY, { ...country, image: img[0] }));
  });
};

const Countries = () => {
  const dispatch = useDispatch();
  const countriesState = useSelector((state) => state.countries);

  useEffect(() => {
    if (countriesState.fetched.length === 0) {
      axios
        .get(`${baseAPI}${getCurrentDate()}`)
        .then((countries) => populateReduxStore(Object.values(countries.data.dates)[0], dispatch));
    }
  }, []);
  return (
    <main>
      <HeroSection />
      <ListOfCountries />
    </main>
  );
};
export default Countries;
