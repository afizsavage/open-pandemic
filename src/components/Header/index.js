import { ImMic } from 'react-icons/im';
import { MdSettings } from 'react-icons/md';
import { useLocation, useNavigate } from 'react-router-dom';
import { BiArrowBack } from 'react-icons/bi';

import SearchBar from './search-bar';

const Header = () => {
  const location = useLocation();
  const navigate = useNavigate();

  return (
    <header className="b-primary flex white-text center-y space-between">
      {location.pathname === '/' && <SearchBar />}
      {location.pathname.includes('/details') && <BiArrowBack onClick={() => navigate('/')} />}
      <span className="head-text">{location.pathname === '/' ? 'Cases' : 'Cases by country'}</span>
      <span>
        <ImMic className="head-icon" />
        <MdSettings className="head-icon" />
      </span>
    </header>
  );
};

export default Header;
