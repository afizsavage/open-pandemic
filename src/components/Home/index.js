import React, { useEffect } from 'react';
import axios from 'axios';
import { useDispatch } from 'react-redux';

import { ADD_COUNTRY, createAction } from '../../redux/countries/index';

const populateReduxStore = (countries, dispatch) => {
  countries.forEach((country) => {
    dispatch(createAction(ADD_COUNTRY, country));
  });
};

const Home = () => {
  const dispatch = useDispatch();
  const baseAPI = 'https://api.covid19tracking.narrativa.com/api/countries';

  useEffect(() => {
    axios.get(baseAPI).then((countries) => populateReduxStore(countries.data.countries, dispatch));
  }, []);
  return <div> Home</div>;
};
export default Home;
