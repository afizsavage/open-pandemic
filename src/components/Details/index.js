/* eslint-disable camelcase */
import { useSelector } from 'react-redux';
import { BsArrowRightCircle } from 'react-icons/bs';

import { SecondHeading } from '../Countries/countries-list';
import HeroSection from '../Countries/hero-section';

const CountryDetails = (object) => {
  const des = Object.values(object)[0];
  const mes = Object.values(des)[0];

  return (
    <>
      <HeroSection />
      <SecondHeading />
      <ul className="details">
        <li className="relative b-fourth flex space white-text">
          <span>Todays Open Cases</span>
          <span>
            {mes.today_new_open_cases.toLocaleString()}
          </span>
          <BsArrowRightCircle className="absolute" />
        </li>
        <li className=" relative b-fourth flex space white-text">
          <span>Todays Confirmed</span>
          <span>
            {mes.today_new_confirmed.toLocaleString()}
          </span>
          <BsArrowRightCircle className="absolute" />
        </li>
        <li className=" relative b-fourth flex space white-text">
          <span>Todays Recovered</span>
          <span>
            {mes.today_new_recovered.toLocaleString()}
          </span>
          <BsArrowRightCircle className="absolute" />
        </li>
        <li className="relative b-fourth flex space white-text">
          <span>Todays Deaths</span>
          <span>
            {mes.today_new_deaths.toLocaleString()}
          </span>
          <BsArrowRightCircle className="absolute" />
        </li>
        <li className="relative b-fourth flex space white-text">
          <span>Total Open Cases</span>
          <span>
            {mes.today_deaths.toLocaleString()}
          </span>
          <BsArrowRightCircle className="absolute" />
        </li>
        <li className=" relative b-fourth flex space white-text">
          <span>Total Confirmed</span>
          <span>
            {mes.today_confirmed.toLocaleString()}
          </span>
          <BsArrowRightCircle className="absolute" />
        </li>
        <li className="relative b-fourth flex space white-text">
          <span>Total Deaths</span>
          <span>
            {mes.today_deaths.toLocaleString()}
          </span>
          <BsArrowRightCircle className="absolute" />
        </li>
        <li className="relative b-fourth flex space white-text">
          <span>Total Recovered</span>
          <span>
            {mes.today_recovered.toLocaleString()}
          </span>
          <BsArrowRightCircle className="absolute" />
        </li>
      </ul>
    </>
  );
};

const Details = () => {
  const state = useSelector((state) => state.countries);
  return (
    <CountryDetails country={state.currentCountry} />
  );
};

export default Details;
