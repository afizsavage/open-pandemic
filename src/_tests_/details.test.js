import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../redux/configure-store';
import { CountryDetails } from '../components/Details';

describe('Test counntry details list compoent', () => {
  it('Renders as expected', () => {
    const testCountry = {
      today_new_open_cases: 7,
      today_new_confirmed: 7,
      today_new_recovered: 9,
      today_new_deaths: 7,
      today_deaths: 5,
      today_open_cases: 3,
      today_confirmed: 87,
      today_recovered: 76,

    };

    const component = renderer.create(
      <Provider store={store}>
        <Router><CountryDetails country={testCountry} /></Router>
      </Provider>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
