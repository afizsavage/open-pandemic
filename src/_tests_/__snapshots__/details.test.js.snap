// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test counntry details list compoent Renders as expected 1`] = `
Array [
  <section
    className="b-secondary full-width"
    id="hero"
  >
    <div
      className="full-height flex center-y flex-end"
    >
      <div
        id="stats"
      >
        <h1
          className="white-text"
        >
          Global
        </h1>
        <span
          className="white-text"
        >
           Cases
        </span>
      </div>
    </div>
  </section>,
  <h2
    className="full-width white-text b-third"
  >
    Stats By Country
  </h2>,
  <ul
    className="details b-fourth"
  >
    <li
      className="relative b-fourth flex space white-text"
    >
      <span>
        Today's Open Cases
      </span>
      <span>
        7
      </span>
      <svg
        className="absolute"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </li>
    <li
      className="relative b-fourth flex space white-text"
    >
      <span>
        Today's Confirmed
      </span>
      <span>
        7
      </span>
      <svg
        className="absolute"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </li>
    <li
      className="relative b-fourth flex space white-text"
    >
      <span>
        Today's Recovered
      </span>
      <span>
        9
      </span>
      <svg
        className="absolute"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </li>
    <li
      className="relative b-fourth flex space white-text"
    >
      <span>
        Today's Deaths
      </span>
      <span>
        7
      </span>
      <svg
        className="absolute"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </li>
    <li
      className="relative b-fourth flex space white-text"
    >
      <span>
        Total Open Cases
      </span>
      <span>
        3
      </span>
      <svg
        className="absolute"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </li>
    <li
      className="relative b-fourth flex space white-text"
    >
      <span>
        Total Confirmed
      </span>
      <span>
        87
      </span>
      <svg
        className="absolute"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </li>
    <li
      className="relative b-fourth flex space white-text"
    >
      <span>
        Total Deaths
      </span>
      <span>
        5
      </span>
      <svg
        className="absolute"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </li>
    <li
      className="relative b-fourth flex space white-text"
    >
      <span>
        Total Recovered
      </span>
      <span>
        76
      </span>
      <svg
        className="absolute"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </li>
  </ul>,
]
`;
